tests/test_discovery.py

```diff
@@ -0,0 +1,96 @@
+import re
+
+from tap_tester import menagerie, connections
+
+from base import TestLinkedinAdsBase
+
+class LinkedinAdsDiscoveryTest(TestLinkedinAdsBase):
+
+    def name(self):
+        return "tap_tester_linedin_ads_discovery_test"
+
+    def test_run(self):
+        """
+        Testing that discovery creates the appropriate catalog with valid metadata.
+
+        • Verify number of actual streams discovered match expected
+        • Verify the stream names discovered were what we expect
+        • Verify stream names follow naming convention
+          streams should only have lowercase alphas and underscores
+        • verify there is only 1 top level breadcrumb
+        • verify primary key(s)
+        • verify that primary keys are given the inclusion of automatic.
+        • verify that all fields have inclusion of available metadata.
+        """
+        streams_to_test = self.expected_streams()
+
+        conn_id = connections.ensure_connection(self)
+
+        found_catalogs = self.run_and_verify_check_mode(conn_id)
+        # Verify stream names follow naming convention
+        # streams should only have lowercase alphas and underscores
+        found_catalog_names = {c['tap_stream_id'] for c in found_catalogs}
+        self.assertTrue(all([re.fullmatch(r"[a-z_]+",  name) for name in found_catalog_names]),
+                        msg="One or more streams don't follow standard naming")
+
+        for stream in streams_to_test:
+            with self.subTest(stream=stream):
+
+                # Verify ensure the catalog is found for a given stream
+                catalog = next(iter([catalog for catalog in found_catalogs
+                                     if catalog["stream_name"] == stream]))
+                self.assertIsNotNone(catalog)
+
+                # collecting expected values
+                expected_primary_keys = self.expected_primary_keys()[stream]
+                expected_replication_keys = self.expected_replication_keys()[stream]
+
+                # automatic fields will be only primary keys as 
+                # replication keys are not automatically replicated
```

Comment:
I think the general expectation is that we should include the replication keys as automatic. Is there a reason that we cannot do that in this tap?

Reply:
Yes, the existing behaviour of that is in such a way that it does not select replication keys as automatically selected in the catalog file.